### PR TITLE
feat: add ensemble retriever backend

### PR DESCRIPTION
**Pull request by palaniappan-r**
Initialized the backend in docker containers. The backend is currently using the Ensemble Retriever architecture. I have updated the README with detailed instructions for installation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,61 @@
...
+```
+- **Step 4**: To run the server,
+```bash
+  cd backend
+  python main.py
+```
+
+## Architecture Overview
+
+Currently, documentation from OpenROAD and OpenROAD-flow-scripts is chunked recursively and embedded into FAISS Vector Databases.
```
Comment (dralabeing) [inline README.md:39]: Are these going to be separated out? OR can be used separately from ORFS.

File: README.md [hunk 2]
```diff
@@ -1 +1,61 @@
...
+```bash
+  cd backend
+  python main.py
+```
+
+## Architecture Overview
+
+Currently, documentation from OpenROAD and OpenROAD-flow-scripts is chunked recursively and embedded into FAISS Vector Databases.  
+
+Upon receiving a query, relevant documents are retrieved and reranked, using [LangChain's Ensemble Retriever](https://python.langchain.com/v0.1/docs/modules/data_connection/retrievers/ensemble/).
```
Comment (dralabeing) [inline README.md:41]: Are you testing hybrid search as well (text+vector indexing) ?

**Review by dralabeing — DISMISSED** [review]
Add a brief introduction to this tool describing its use model.

File: README.md (hunk 2 above)
Comment (palaniappan-r) [reply README.md:41]: I'm implementing a version of hybrid search in the [branch](https://github.com/The-OpenROAD-Project/ORAssistant/tree/ensemble-retriever-update).

File: README.md (hunk 1 above)
Comment (palaniappan-r) [reply README.md:39]: They are not separated currently, but we are exploring options where the architecture can selectively query different data sources depending on the input query.

**Review by luarss — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -58,4 +58,9 @@ By default,
 
 The backend will then be hosted at [http://0.0.0.0:8000](http://0.0.0.0:8000). 
 
-Open [http://0.0.0.0:8000/docs](http://0.0.0.0:8000/docs) for the API docs.
\ No newline at end of file
+Open [http://0.0.0.0:8000/docs](http://0.0.0.0:8000/docs) for the API docs.
+
```
Comment (dralabeing) [inline README.md:62]: Change to an introduction as follows (replace the description below for use cases):
The OpenROAD chat assistant aims to provide easy and quick access to information regarding tools, responses to questions and commonly occurring problems in OpenROAD and its native flow-OpenROAD-flow-scripts.

The current architecture is based on retrieval techniques from OpenROAD documentation and other online data sources. We aim to continuously improve the architecture and the associated the dataset to improve accuracy, coverage  and robustness.

Comment (palaniappan-r) [reply README.md]: Done, please check now.

**Review by luarss — APPROVED** [review]

**Review by luarss — APPROVED** [review]

**Pull request merged**